hw-event-proxy/proxy.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021 The Cloud Native Events Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	hwevent "github.com/redhat-cne/sdk-go/pkg/hwevent"
+
+	"github.com/redhat-cne/sdk-go/pkg/channel"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/pb"
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/restclient"
+	"google.golang.org/grpc"
+
+	v1amqp "github.com/redhat-cne/sdk-go/v1/amqp"
+	v1hwevent "github.com/redhat-cne/sdk-go/v1/hwevent"
+	v1pubsub "github.com/redhat-cne/sdk-go/v1/pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+// SCConfiguration simple configuration to initialize variables
+type SCConfiguration struct {
+	EventInCh  chan *channel.DataChan
+	EventOutCh chan *channel.DataChan
+	CloseCh    chan struct{}
+	APIPort    int
+	APIPath    string
+	PubSubAPI  *v1pubsub.API
```

Comment:
Won't be using PubSubAPI, you will rely on Rest call to Sidecar.

Reply:
check consumer example
https://github.com/redhat-cne/cloud-event-proxy/blob/c9314354d5b8d905a700837d8f1c514da6f9a5bf/examples/consumer/main.go#L167